src/Assignments/ProblemDetails.tsx

```diff
@@ -109,17 +111,19 @@ export const ProblemDetails: React.FC<ProblemDetailsProps> = ({
                         >
                             {(currentMoment: moment.Moment) => {
                                 let message = '';
-                                if (currentMoment.isBefore(endDate) && (_.isNil(attemptsRemaining) || attemptsRemaining > 0)) {
+                                if (currentMoment.isBefore(endDate) && (_.isNil(attemptsRemaining) || (attemptsRemaining > 0 && !isClosed))) {
                                     message = `Due ${endDate.fromNow()}`;
                                 } else if (currentMoment.isBefore(deadDate)) {
                                     message = `Partial credit expires ${deadDate.fromNow()}`;
                                 } else if (currentMoment.isBefore(solutionsMoment)) {
                                     message = `Solutions available ${solutionsMoment.fromNow()}`;
-                                } else if (isVersionedAssessment && (attemptsRemaining === 0 || isClosed)) {
-                                    message = 'is completed';
-                                } else {
+                                } else if (currentMoment.isAfter(solutionsMoment)) {
                                     message = (isVersionedAssessment) ? 'Time expired for this version' : 'Past due';
+                                    setAttemptsRemaining && setAttemptsRemaining(0);
```

Comment:
Is this you using react lifecycle to use effect rather than just calling the method directly

Does this disrupt homework at all? I'd assume it calls a rerender but then nothing happens so it is fun

Reply:
I could do `isVersionedAssessment && setAttemptsRemaining?.(0)`

Reply:
Yeah, I don't think it is a big deal